api/tests/integration/infrastructure/repositories/campaign-profiles-collection-participation-summary-repository_test.js

```diff
@@ -76,11 +88,89 @@ describe('Integration | Repository | Campaign Profiles Collection Participation
       await databaseBuilder.commit();
   
       // when
-      const results = await CampaignProfilesCollectionParticipationSummaryRepository.findPaginatedByCampaignId(campaignId);
+      const results = await campaignProfilesCollectionParticipationSummaryRepository.findPaginatedByCampaignId(campaignId);
       const names = results.data.map((result) => result.firstName);
   
       // then
       expect(names).exactlyContainInOrder(['Arthur', 'Yvonnick', 'Estelle', 'Benjamin', 'Lise']);
     });
+
+    describe('when a participant has shared the participation to the campaign', () => {
+      let campaignParticipation;
+    
+      beforeEach(async () => {
+        const createdAt = new Date('2018-04-06T10:00:00Z');
+        const userId = 999;
+
+        campaignParticipation = { id: 1, campaignId, isShared: true, sharedAt };
+        createCampaignParticipation({ id: userId }, campaignParticipation);
+
+        databaseBuilder.factory.buildKnowledgeElement({
+          status: 'validated',
+          competenceId: competences[0].id,
+          skillId: skills[0].id,
+          earnedPix: 40,
+          userId,
+          createdAt,
+        });
+      
+        databaseBuilder.factory.buildKnowledgeElement({
+          status: 'validated',
+          competenceId: competences[1].id,
+          skillId: skills[2].id,
+          earnedPix: 6,
+          userId,
+          createdAt,
+        });
+
+        await databaseBuilder.commit();
+      });
+
+      it('should return the certification profile info and pix score', async () => {
+        // when
+        const results = await campaignProfilesCollectionParticipationSummaryRepository.findPaginatedByCampaignId(campaignId);
+    
+        // then
+        expect(results.data[0].sharedAt).to.deep.equal(sharedAt);
+        expect(results.data[0].pixScore).to.equal(46);
+        expect(results.data[0].certifiable).to.equal(false);
+        expect(results.data[0].certifiableCompetencesCount).to.equal(1);
+      });
+    });
   });
 });
+
+const buildAirtableData = () => {
```

Comment:
Je comprends pas comment ça peut fonctionner. Pour moi le hoisting fonctionnait que avec le mot-clé `function`. Est ce que quelqu'un peut m'expliquer ?

Reply:
Ça fonctionne car `buildAirtableData` est évalué avant sont utilisation (quand la fonction `beforeEach()` est exécutée.

L'exemple suivant ne fonctionne pas car la fonction est exécuté avant que `test` soit évalué :
```js
hello() // executé avant l'évaluation de test

function hello() {
  console.log(test)
}

const test = "toto"
```

Mais l'exemple suivant fonctionne car la fonction est exécuté après que `test` soit évalué :
```js

function hello() {
  console.log(test)
}

const test = "toto"

hello() // executé après l'évaluation de test
```

Dans le cadre des tests unitaire les fonctions `describe`, `it`, ... sont éxécutées après l'évaluation complète de chaque fichier de test.

Reply:
👍 pour l'explication. Après, si on regarde le code de Pix on voit que le standard est plutôt d'utiliser des déclarations de fonctions pour les fonctions déclarées au niveau module :
```js
function buildAirtableData() {
	…
}
```